src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -68,6 +68,7 @@ public static partial class MarshalHelpers
                 case MarshallerKind.BlittableStruct:
                 case MarshallerKind.Decimal:
                 case MarshallerKind.VoidReturn:
+                case MarshallerKind.ComInterface:
```

Comment:
This should be some kind of a pointer, not `type`. E.g. `return context.GetWellKnownType(WellKnownType.IntPtr);`

(This is the native type we marshal into. Leaving this at `type` would hit RyuJIT asserts added in dotnet/runtime#35026 because `type` is a GC pointer. Once marshalling is done, this is an unmanaged pointer.)